core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -411,10 +439,14 @@ Builder scanMetrics(ScanMetrics newScanMetrics) {
       return this;
     }
 
-    DeleteFileIndex build() {
+    private Collection<DeleteFile> planFiles() {
+      if (deleteFiles != null) {
+        return deleteFiles;
```

Comment:
Should this apply the `minSequenceNumber` filter? Seems like that would be more consistent.

Reply:
This was an early prototype. I will change how this works. All other `filterXXX` methods should still work.
I created #8157 with changes to the delete index alone.